### PR TITLE
Added Wrapper for NewArrayBufferWithContents

### PR DESCRIPTION
**Pull request by Redfire75369**
Removing the two wrapped array buffer functions from the bindings also shortens imports (`mozjs::jsapi::<function>` as compared to `mozjs::jsapi::glue::<function>`).

### CODE REVIEW

**Review by jdm — APPROVED** [review]

**Comment by sagudev** [comment]
This will need rebase after #483 lands, so let's wait for it to land.

**Comment by Redfire75369** [comment]
Rebased on latest, with version bump accordingly.

**Pull request merged**
